lgc/builder/ImageBuilder.cpp

```diff
@@ -753,12 +753,15 @@ Value *ImageBuilder::CreateImageSampleConvertYCbCr(Type *resultTy, unsigned dim,
   Value *ycbcrSamplerDesc = CreateExtractValue(convertingSamplerDesc, 0);
 
   // Extract YCbCr meta data
-  const SamplerYCbCrConversionMetaData yCbCrMetaData = {
-      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(4)))->getZExtValue()),
-      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(5)))->getZExtValue()),
-      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(6)))->getZExtValue()),
-      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(7)))->getZExtValue()),
-  };
+  SamplerYCbCrConversionMetaData yCbCrMetaData;
+  yCbCrMetaData.word0.u32All =
+      static_cast<unsigned>(dyn_cast<ConstantInt>(CreateExtractElement(ycbcrSamplerDesc, getInt64(4)))->getZExtValue());
```

Comment:
Use cast<> not dyn_cast<>, as that asserts that you are relying on it being that type.